lesson12.md

```diff
@@ -0,0 +1,412 @@
+```
+
+That's good enough!
+But would it be possible to store all this data into an array?
+<!-- .element: class="fragment"  -->
+
+---
+
+### Recap: data types
+
+```js
+
+let me = [39, "Carlo", "Trimarchi", "Italian", "Messina",
+          "green", "black", false, true, hour > 20,
+          ['reading', 'music', 'guitar'],
+          ['pizza', 'pasta', 'lasagna']];
+```
+
+It is possible, but it's a mess.
+<!-- .element: class="fragment"  -->
+
+---
+
+### Objects
+
+An **object**, in Javascript, is a group (or a list) of *key* / *value* pairs
+
+![Pears](images/pears.jpg) <!-- .element width="300" style="display: block; margin: 0 auto; width: 300px;" -->
+
+
+---
+
+### Objects
+
+Example:
+
+```js
+let person = {
+  firstName: 'Carlo',
+  lastName: 'Trimarchi',
+  age: 39
+};
+```
+
+Can you guess how we can create an object?
+
+---
+
+### Objects
+
+To create an object we use curly braces `{}`.
+```js
+let emptyObject = {};
+```
+
+We put any key/value pair inside the `{}`
+
+```js
+let person = {
+  name: 'Carlo'
+  // the key is 'name', the value is 'Carlo'
+};
+```
+
+---
+
+### Objects
+
+We can think of key/value pairs as words in a dictionary
+
+![Pears](images/dictionary.jpg) <!-- .element width="500" style="display: block; margin: 0 auto;" -->
+
+* a word is the **key**
+* the definition is the **value**
+
+---
+
+### Objects
+
+key/value pairs are also referred to as **properties**
+
+```js
+let person = {
+  firstName: 'Carlo', // comma here
+  age: 39, // also here
+  isVegetarian: true // this is the last one, no comma
+};
+```
+* can you list all the properties in this **person** object?
+* and its keys?
+* and its values?
+
+We can add as many properties as we want.
+<!-- .element: class="fragment"  -->
+
+---
+
+### Objects: examples
+
+```js
+
+let book = {
+  title: 'The Lord of the Rings',
+  author: 'J. R. R. Tolkien',
+  publicationYear: 1954,
+  pages: 1216
+};
+
+```
+
+Can you think of other things we can represent with objects?
+<!-- .element: class="fragment"  -->
+
+---
+
+### Objects: examples
+
+* movies
+<!-- .element: class="fragment"  -->
+* passport / identity card
+<!-- .element: class="fragment"  -->
+* cooking recipe
+<!-- .element: class="fragment"  -->
+* product in an e-commerce
+<!-- .element: class="fragment"  -->
+* much more...
+<!-- .element: class="fragment"  -->
+
+---
+
+### Objects: exercise
+
+Try it yourself:
+
+create an object called "`me`" containing the following information about you:
+* name
+* age
+* hair color
+* eye color
+
+Log your object to the console
+
+---
+
+### Objects vs arrays
+
+```js
+let meArray = [39, "Carlo", "Trimarchi", "Italian", "Messina",
+          "green", "black", false, true,
+          ['reading', 'music', 'guitar'],
+          ['pizza', 'pasta', 'lasagna']];
+
+let me = {
+  age: 39,
+  firstName:  "Carlo",
+  lastName:  "Trimarchi",
+  nationality:  "Italian",
+  cityOfBirth:  "Messina",
+  favouriteColour:  "green",
```

Comment:
Shall we use British English or US? Color vs. Colour

Reply:
oh, right. I think I'm used to write "colour", but in programming is definitely better to use "color"